server/implementation/src/main/java/io/smallrye/graphql/execution/datafetcher/DefaultDataFetcher.java

```diff
@@ -48,10 +53,17 @@ public DefaultDataFetcher(Operation operation, Config config) {
     public CompletionStage<List<T>> load(List<K> keys, BatchLoaderEnvironment ble) {
         Object[] arguments = batchLoaderHelper.getArguments(keys, ble);
         final ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+        final SmallRyeContext context = ble.getContext();
+
         ThreadContext threadContext = ThreadContext.builder().build();
-        return threadContext
-                .withContextCapture(
-                        CompletableFuture.supplyAsync(() -> (List<T>) reflectionHelper.invokePrivileged(tccl, arguments),
-                                threadContext.currentContextExecutor()));
+        try {
+            SmallRyeContext.setContext(context);
+            return threadContext
+                    .withContextCapture(
+                            CompletableFuture.supplyAsync(() -> (List<T>) reflectionHelper.invokePrivileged(tccl, arguments),
+                                    threadContext.currentContextExecutor())); // Infrastructure.getDefaultExecutor ????
```

Comment:
I think you can remove the Infrastructure.getDefa....????? comment. That in Mutiny specific, And Mutiny has it's own datafetcher.